[spacing] Export spacing and add docs

wfro opened the pull request:
### What

Exports the spacing module from snacks, and adds a section to the style guide with a few examples.
![screen shot 2017-09-19 at 3 15 34 pm](https://user-images.githubusercontent.com/6454987/30617989-6af8e4e8-9d4d-11e7-8f0a-9e5964a9dcc3.png)

I didn't include any runnable examples, but certainly open to doing so.

dcocchia commented on docs/spacing.md:
```diff
@@ -0,0 +1,35 @@
+  // Padding/margin along one dimension
+  ...spacing.PADDING_Y_XL              // => { paddingTop: 64, paddingBottom: 64 }
+  ...spacing.MARGIN_X_LG,              // => { marginLeft: 32, marginRight: 32 }
+}
```
spacing also works with `...spacing.LEFT_8` which outputs `{ left: 8 }`. Can you add that to the examples?

wfro commented:
@dcocchia added a section for positioning, and also added a few exports to the spacing module itself.

Turns out we weren't actually exporting things like `PADDING_X_MD` (whoops!), so now those are there as well.

Is it worth it to add a test or two to this file at this point?

dcocchia commented:
yeah, tests 👍

wfro commented:
Added a few tests (they don't cover every case, but enough to give me confidence), and also adjusted the docs to match how positioning actually works, so it's `LEFT_SM` instead of `LEFT_16`.

wfro commented:
@dcocchia Also question - do you do new releases by hand for now? I think the process is

* yarn build
* yarn styleguide:build
* bump version in package.json

Should I do this in a separate commit after this makes it in? Or have you been doing the releases for now

dcocchia commented:
I've been doing releases manually, yeah. I do them on a new branch named after the release ex - https://github.com/instacart/Snacks/tree/0.0.43

- create new branch named after release number (ex 0.0.44)
- yarn build
- yarn styleguide:build
- bump version in package.json
- create PR from branch
- Make sure all tests are passing, etc
- Merge with master
- On master, run `npm publish`
